ibis/tests/expr/test_struct.py

```diff
@@ -41,7 +42,30 @@ def test_struct_getattr():
     assert isinstance(expr.a, ir.IntegerValue)
     assert expr.a.get_name() == "a"
     with pytest.raises(AttributeError, match="bad"):
-        expr.bad  # # noqa: B018
+        expr.bad  # noqa: B018
+
+
+def test_struct_getitem():
+    expr = ibis.struct({"a": 1, "b": 2})
+    assert isinstance(expr.a, ir.IntegerValue)
+    assert expr["a"].get_name() == "a"
+
+    with pytest.raises(TypeError, match="Deferred"):
+        expr[_.A]
+
+    with pytest.raises(com.FieldsNotFoundError, match="A") as excinfo:
+        expr["A"]
+    assert isinstance(excinfo.value, AttributeError)
+
+    assert excinfo.value.container is expr
```

Comment:
You're using `container` in tests here, but that doesn't seem like a good reason to store an expression as an attribute on an exception.